packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx

```diff
@@ -21,69 +34,343 @@ import styled from 'styled-components';
+                        {formatMessage({
+                          id: 'content-releases.header.actions.delete',
+                          defaultMessage: 'Delete',
+                        })}
+                      </Typography>
+                    </PopoverButton>
+                  </Flex>
+                  <ReleaseInfoWrapper
+                    direction="column"
+                    justifyContent="center"
+                    alignItems="flex-start"
+                    gap={1}
+                    padding={5}
+                  >
+                    <Typography variant="pi" fontWeight="bold">
+                      {formatMessage({
+                        id: 'content-releases.header.actions.created',
+                        defaultMessage: 'Created',
+                      })}
+                    </Typography>
+                    <Typography variant="pi" color="neutral300">
+                      {formatMessage(
+                        {
+                          id: 'content-releases.header.actions.created.description',
+                          defaultMessage:
+                            '{number, plural, =0 {# {unit}s} one {# {unit}} other {# {unit}s}} ago by {createdBy}',
+                        },
+                        { number: -value, unit, createdBy }
+                      )}
+                    </Typography>
+                  </ReleaseInfoWrapper>
+                </Popover>
+              )}
+              <Button size="S" variant="tertiary">
+                {formatMessage({
+                  id: 'content-releases.header.actions.refresh',
+                  defaultMessage: 'Refresh',
+                })}
+              </Button>
+              <Button size="S" disabled={true} variant="default">
+                {formatMessage({
+                  id: 'content-releases.header.actions.release',
+                  defaultMessage: 'Release',
+                })}
+              </Button>
+            </Flex>
+          }
+        />
+      </Box>
+
+      {children}
+    </Main>
+  );
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * ReleaseDetailsBody
+ * -----------------------------------------------------------------------------------------------*/
+const ReleaseDetailsBody = () => {
+  const { formatMessage } = useIntl();
+  const { releaseId } = useParams<{ releaseId: string }>();
+  const [{ query }] = useQueryParams<GetReleaseActionsQueryParams>();
+
+  const { isLoading, isFetching, isError, data } = useGetReleaseActionsQuery({
+    ...query,
+    releaseId,
+  });
+
+  if (isLoading) {
+    return (
+      <ContentLayout>
+        <LoadingIndicatorPage />
+      </ContentLayout>
+    );
+  }
+
+  if (isError) {
+    return (
+      <ContentLayout>
+        <AnErrorOccurred />
+      </ContentLayout>
+    );
+  }
+
+  const zeroActions = data?.data?.length === 0;
+
+  if (zeroActions) {
```

Comment:
I think you need to check both you have a release and the release is not so you don't have to keep supporting fallbacks below. For example here if you did something like

```tsx
const releaseActions = data?.data;
  if (!releaseActions || !releaseActions.length) {
```

Then you wouldn't have to constantly support fallbacks like:

```tsx
rows={
  data?.data.map((item) => ({
    ...item,
    id: Number(item.entry.id),
  })) || []
}
colCount={data?.data?.length || 0}
```

If we are done loading, and there is no error, we should have an array not undefined